File: src/lib/email.ts
import nodemailer from "nodemailer";

type SendEmailOptions = {
  to: string;
  subject: string;
  html: string;
};

export function isEmailConfigured(): boolean {
  return Boolean(
    process.env.SMTP_HOST &&
      process.env.SMTP_USER &&
      process.env.SMTP_PASS &&
      process.env.SMTP_FROM
  );
}

export async function sendEmail({ to, subject, html }: SendEmailOptions) {
  if (!isEmailConfigured()) {
    return { sent: false as const };
  }

  const transporter = nodemailer.createTransport({
    host: process.env.SMTP_HOST,
    port: Number(process.env.SMTP_PORT || 587),
    secure: process.env.SMTP_SECURE === "true",
    auth: {
      user: process.env.SMTP_USER,
      pass: process.env.SMTP_PASS,
    },
  });

  await transporter.sendMail({
    from: process.env.SMTP_FROM,
    to,
    subject,
    html,
  });

  return { sent: true as const };
}

export function getAppUrl() {
  return process.env.APP_URL || "http://localhost:3000";
}

File: mizhara-backend/lib/email.go
package lib

import (
	"fmt"
	"net/smtp"
	"os"
	"strconv"
)

func IsEmailConfigured() bool {
	return os.Getenv("SMTP_HOST") != "" &&
		os.Getenv("SMTP_USER") != "" &&
		os.Getenv("SMTP_PASS") != "" &&
		os.Getenv("SMTP_FROM") != ""
}

type EmailResult struct {
	Sent bool `json:"sent"`
}

func SendEmail(to, subject, html string) (EmailResult, error) {
	if !IsEmailConfigured() {
		return EmailResult{Sent: false}, nil
	}

	host := os.Getenv("SMTP_HOST")
	port, _ := strconv.Atoi(os.Getenv("SMTP_PORT"))
	if port == 0 {
		port = 587
	}
	from := os.Getenv("SMTP_FROM")
	user := os.Getenv("SMTP_USER")
	pass := os.Getenv("SMTP_PASS")

	msg := []byte(fmt.Sprintf("To: %s\r\nSubject: %s\r\nMIME-Version: 1.0\r\nContent-Type: text/html; charset=UTF-8\r\n\r\n%s", to, subject, html))
	auth := smtp.PlainAuth("", user, pass, host)
	addr := fmt.Sprintf("%s:%d", host, port)
	err := smtp.SendMail(addr, auth, from, []string{to}, msg)
	if err != nil {
		return EmailResult{Sent: false}, err
	}
	return EmailResult{Sent: true}, nil
}

func GetAppURL() string {
	if u := os.Getenv("APP_URL"); u != "" {
		return u
	}
	return "http://localhost:5173"
}
